### PR TITLE
Adding STFT from ONNX to Ivy Experimental Files

### LINKED ISSUE #14668: STFT

### PR DESCRIPTION
**Pull request by KevinUnadkat**
Closes #14668

### CODE REVIEW

**Comment by vedpatwardhan** [comment]
Hey @KevinUnadkat, just looked through the PR, the problem with the `jax` backend is with the `axis` argument, for which we've added the default value as `None`, but `jax.scipy.signal.stft` has a default value of `-1`. Basically you'd need to infer the `axis` from the `signal` and other relevant arguments of the `ivy` function and pass that inferred `axis` to `jax.scipy.signal.stft` if `axis` is `None`. 
Also, secondly you could add something like,
```
stft.mixed_backend_wrappers = {
    "to_add": (
        "inputs_to_native_arrays",
        "outputs_to_ivy_arrays",
    ),
    "to_skip": ("inputs_to_ivy_arrays",),
}   
```
instead of `stft.mixed_function=True` which was the old method (more info [here](https://unify.ai/docs/ivy/overview/deep_dive/function_types.html#mixed-functions))
Hope that helps :smile:

**Comment by zhumakhan** [comment]
Hi @KevinUnadkat, could you please add test for this?

File: ivy/data_classes/array/experimental/layers.py [hunk 1]
```diff
@@ -958,7 +958,7 @@ def fft2(
         out: Optional[ivy.Array] = None,
     ) -> ivy.Array:
         """
-        Compute the 2-dimensional discrete Fourier Transform.
+        Compute the discrete Fourier transform of input.
```
Comment (zhumakhan) [inline ivy/data_classes/array/experimental/layers.py:961]: Hey, what is the reason that this description is updated?

File: ivy_tests/test_ivy/test_functional/test_experimental/test_nn/test_layers.py [hunk 2]
```diff
@@ -1268,3 +1268,106 @@ def test_rfftn(
...
+@handle_test(
+    fn_tree="functional.ivy.experimental.stft",
+    dtype_and_x=helpers.dtype_and_values(
+        available_dtypes=helpers.get_dtypes("float"),
+    ),
+    ground_truth_backend="numpy",
+    test_gradients=st.just(False),
+    stft_args=stft_arguments(),
+)
+def test_stft(
```
Comment (zhumakhan) [inline ivy_tests/test_ivy/test_functional/test_experimental/test_nn/test_layers.py:1325]: Hey, could you please fix tests: `hypothesis.errors.InvalidArgument: max_value=(2, 2) must be a real number.`

**Review by zhumakhan — CHANGES_REQUESTED** [review]

**Comment by zhumakhan** [comment]
Hey, this PR has been updated a lot since last comments I left. Could you please respond to those comments with how latest changes addressed them?

File: ivy_tests/test_ivy/test_functional/test_experimental/test_nn/test_layers.py [hunk 3]
```diff
@@ -1263,6 +1263,110 @@ def test_rfftn(
...
+    else:
+        n_fft = tuple(
+            draw(st.integers(min_value=1, max_value=256))
+            for _ in range(2))
+    
+    hop_length = draw(st.integers(min_value=1, max_value=256))
+    axis = draw(st.integers(min_value=0))
+    onesided = draw(st.booleans())
+    fs = 1.0
+    window = draw(helpers.get_dtypes("float", full=False)))
```
Comment (zhumakhan) [inline ivy_tests/test_ivy/test_functional/test_experimental/test_nn/test_layers.py:1283]: Parentheses do not match here 2x( and 3x)

**Review by zhumakhan — CHANGES_REQUESTED** [review]

**Review by vedpatwardhan — COMMENTED** [review]
Given that the focus is shifted from adding new functions to fixing and optimizing those used in popular models, I think we should close this PR now as there's already another `stft` function on `main` 😄

**Review by vedpatwardhan — COMMENTED** [review]
I believe @KevinUnadkat is still working on this, feel free to request a review once you're done with the changes. Thanks @KevinUnadkat 😄

**Comment by aibenStunner** [comment]
Hi @KevinUnadkat,

Thanks for the PR!

However, it looks like it has been inactive for a while.

Therefore, I’ve simply closed it for now. 🙂

Please feel free to submit other PRs based on our [Open Tasks](https://lets-unify.ai/ivy/overview/contributing/open_tasks.html).

Thanks!
